bibliopixel/project/project.py

```diff
@@ -17,10 +17,11 @@ def _make_layout(layout, driver=None, drivers=None, maker=None):
     if driver is None and drivers is None:
         raise ValueError('Projects has no driver or drivers section')
 
+    coord_map = layout.get('coordMap', None)
+    layout.pop('coordMap', None)
```

Comment:
While I know this doesn't cause a problem in the code as it is, I've avoided actually changing the project dictionary that is passed in (which you are doing, indirectly, with that `pop`).

The reason is simple - because I don't like it when calling exactly the same method twice with the same data gives different results, and I don't like it when my methods rewrite my configuration dictionaries.

This is a lot of the reason that I go through these fancies with using `**` and a method call to unpack dictionaries and snip out individual, optional parameters.

Which now I write this might have been more trouble that it should have been.

So two options here:

1. `copy.deepcopy` the project when we first see it in this file, or
2. Figure out a way by using argument passing not to change the originals.

I'm thinking that 1. might actually be better, which is a shame, since I started with that and then decided to get too clever.  :-D

If you did 1. I might go back and simplify some of this dumbshit afterwards.